internal/model/analysis.go

```diff
@@ -213,6 +217,41 @@ func (this *PairwiseKey) Course() string {
 	return courseID
 }
 
+func (this PairwiseAnalysisMap) MarshalJSON() ([]byte, error) {
+	temp := make(map[string]*PairwiseAnalysis, len(this))
+
+	for key, analysis := range this {
+		keyString := key.String()
+
+		temp[keyString] = analysis
+	}
+
+	return json.Marshal(temp)
+}
+
+func (this *PairwiseAnalysisMap) UnmarshalJSON(data []byte) error {
+	temp := make(map[string]*PairwiseAnalysis)
+	err := json.Unmarshal(data, &temp)
+	if err != nil {
+		return err
+	}
+
+	results := make(PairwiseAnalysisMap, len(temp))
+	for text, analysis := range temp {
+		keyParts := strings.Split(text, PAIRWISE_KEY_DELIM)
+		if len(keyParts) != 2 {
+			return fmt.Errorf("Invalid PairwiseKey: '%s'.", text)
+		}
+
+		pairwiseKey := NewPairwiseKey(keyParts[0], keyParts[1])
+		results[pairwiseKey] = analysis
+	}
+
+	*this = results
+
+	return nil
+}
```

Comment:
I think you can instead get away with implementing `encoding.TextMarshaler` since this is for map keys,
and that should be easier to implement than raw JSON marshaling.
See https://pkg.go.dev/encoding/json#Marshal
(search for TextMarshaler in this section).